### ISSUE TITLE
$([MSBuild]::MakeRelative()) returns different values on 16.10 than older version

### ISSUE THREAD

**Issue by yodovicz** [dotnet/msbuild#6493]
### Issue Description
Newest version of VisualStudio (MSBuild 16.10) return different values on msbuild property function MakeRelative() with paths ending backslash.

### Steps to Reproduce
code: $([MSBuild]::MakeRelative('c:\windows\\', 'c:\windows\system32\\'))
MSBuild 16.10 returns:  'system32'
MSBuild 16.9 and older returns: 'system32\\'

### Expected Behavior
according to https://docs.microsoft.com/en-us/dotnet/api/system.uri.makerelativeuri?view=net-5.0
it should returns 'system32\\'

### Actual Behavior
returns 'system32'

### Analysis
MSBuild unittests does not contains paths ending backslash
https://github.com/dotnet/msbuild/blob/main/src/Shared/UnitTests/FileUtilities_Tests.cs#L86

### Versions & Configurations
16.10.0.26302

**Comment by pmisik** [comment]
Here is sample project to verify behavior has changed
```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Build">
  <Target Name="Build">
    <PropertyGroup>
	<_RelativeDir>$([MSBuild]::MakeRelative('c:\windows\', 'c:\windows\system32\'))</_RelativeDir>
    </PropertyGroup>
    <Message Text="MakeRelative('c:\windows\', 'c:\windows\system32\')): $(_RelativeDir)" />
    <Error Condition="!$(_RelativeDir.EndsWith('\'))" Text="_RelativeDir='$(_RelativeDir)' does not ends with backslash" />
  </Target>
</Project>
```

**Comment by BrunoJuchli** [comment]
This change causes execution of our application to fail. We're using [System.Data.SQLite.Core](https://www.nuget.org/packages/System.Data.SQLite.Core/) which now fails to include the SQLite.Interop.dll as content.
When executing the application we get the following exception:

>System.DllNotFoundException : Unable to load DLL 'SQLite.Interop.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
at System.Data.SQLite.UnsafeNativeMethods.sqlite3_config_none(SQLiteConfigOpsEnum op)
at System.Data.SQLite.SQLite3.StaticIsInitialized()
at System.Data.SQLite.SQLiteLog.Initialize(String className)
at System.Data.SQLite.SQLiteConnection..ctor(String connectionString, Boolean parseViaFramework)
at System.Data.SQLite.SQLiteConnection..ctor()

**Comment by dsparkplug** [comment]
The following line appears to be the cause of this issue: 
https://github.com/dotnet/msbuild/pull/6311/files#diff-b62db5f7e7f90841af8ef58195eb4318cabc0742747ba3f2e97f10cfa21d6658R1081

**Comment by pmisik** [comment]
Related report
https://developercommunity.visualstudio.com/t/MSBuild::MakeRelative-in-VS-2019-Versi/1435347

**Comment by marcinzi** [comment]
We're hitting downstream build issues on SQLite.interop.dll as reported by Bruno
with no reasonable workaround other than downgrading build agent pool to self-hosted

**Comment by dsparkplug** [comment]
Added a new pull request to replace this one https://github.com/dotnet/msbuild/pull/6508

**Comment by nzbart** [comment]
I know +1 comments are frowned upon, but at the same time they seem to be the only way to get things prioritised. Therefore:

+1

**Comment by teh13th** [comment]
+1 to this problem.

**Comment by jeremy-morren** [comment]
We are also having the problem with System.Data.SQLite.  This is a blocking issue, since no builds work.

**Comment by cmichaelingram** [comment]
Bit us too.  Blocking as no builds work that use the SQLite nuget packages.

**Comment by dsparkplug** [comment]
A pull request https://github.com/dotnet/msbuild/pull/6513 to fix this has now been merged into the vs16.10 branch and is listed in the servicing fixes for v16.10.2 https://github.com/dotnet/msbuild/pull/6537

**Comment by rainersigwald** [comment]
16.10.2 is now released for Visual Studio customers.
